src/GUI/GraphPanel.java

```diff
@@ -165,7 +175,25 @@ private void drawGrid(Graphics2D g2d, int width, int height, double step) {
         }
     }
 
-    private void drawFunctions(Graphics2D g2d, int width) {
+    private void drawParametricFunction(Graphics2D g2d, ParametricExpression expression) {
+        GeneralPath path = new GeneralPath();
+        double sampleLength = (expression.upperBound - expression.lowerBound) / expression.numSamples;
+
+        Vector2D startPosition = toScreenCoordinates(expression.evaluate(expression.lowerBound));
+        path.moveTo(startPosition.x, startPosition.y);
+
+        for(int i = 1; i <= expression.numSamples; i++) {
+            double t = expression.lowerBound + i * sampleLength;
+            Vector2D position = toScreenCoordinates(expression.evaluate(t));
+            path.lineTo(position.x, position.y);
+        }
+
+        g2d.setStroke(new BasicStroke(2.0f));
+        g2d.setColor(Color.WHITE);
```

Comment:
If there is now a drawParametricFunction differentiating in the drawPolynomialFunctions is not needed anymore. (Will fix that in a sec)

Reply:
Done